### PR TITLE
SILVER-270 - Add support for GitHub Apps

### PR DESCRIPTION
**Pull request by moltenice**
The action now supports a client ID and private key as authentication. This is usually what's needed for authenticating as a GitHub App.

I have re-run and updated all tests, and I believe I have removed every instance of sensitive information (secrets etc.) I could find when I re-ran VCR.

### CODE REVIEW

File: lib/config/github_api_config.rb [hunk 1]
```diff
@@ -6,7 +6,14 @@
 # Sets the github client and event payload
 class GithubApiConfig
   def initialize
-    @client = Octokit::Client.new(access_token: ENV.fetch('ISSUE_TOKEN'))
+    @client = if ENV['ISSUE_TOKEN'] && !ENV['ISSUE_TOKEN'].empty?
+                Octokit::Client.new(access_token: ENV['ISSUE_TOKEN'])
+              else
```
Comment (addersuk) [inline lib/config/github_api_config.rb:11]: In v2 we do not want to support PATs

**Review by addersuk — CHANGES_REQUESTED** [review]
This is doing auth to impresionate someone, we need an app installation via a JWT https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app

**Comment by moltenice** [comment]
> This is doing auth to use oauth client credentials, we need an app installation via a JWT https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app

I am pretty sure this internally uses JWK to do so: https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app#using-the-octokitjs-sdk-to-authenticate-as-a-github-app

File: README.md [hunk 2]
```diff
@@ -40,8 +40,9 @@ specific PR.
          - name: Deploy Status Action
            uses: simplybusiness/deploy-status-action@v0.4.1
```
[the remark this answers is not in the document]
Comment (moltenice) [reply README.md:41]: @addersuk I was thinking of changing this after this PR but sure.

[the remark this answers is not in the document]
Comment (addersuk) [reply README.md]: So either don't add the new parameters, or bump to v2

[the remark this answers is not in the document]
Comment (moltenice) [reply README.md]: Bumped to v2

**Review by addersuk — APPROVED** [review]

**Pull request merged**